src/__tests__/index.test.ts

```diff
@@ -31,160 +51,200 @@ afterAll(() => {
+    useRedis = false;
+    await clearStats();
   });
-  expect(data).toBeTruthy();
-  expect(axiosCachingDns.getStats().dnsEntries).toBe(1);
-  expect(axiosCachingDns.getStats().misses).toBe(1);
-
-  const dnsCacheEntries = axiosCachingDns.getDnsCacheEntries();
-  expect(Array.isArray(dnsCacheEntries)).toBeTruthy();
-  expect(Array.isArray(dnsCacheEntries[0].ips)).toBeTruthy();
-  expect(dnsCacheEntries[0].ips[0]).toBeTruthy();
-  expect(dnsCacheEntries[0].host).toBeTruthy();
-  expect(dnsCacheEntries[0].lastUsedTs).toBeTruthy();
-  expect(dnsCacheEntries[0].updatedTs).toBeTruthy();
-});
 
-test('query google caches and after idle delay uncached', async () => {
-  const resp = await axiosClient.get('http://google.com');
-  expect(resp.data).toBeTruthy();
-  expect(axiosCachingDns.config.cache?.get('google.com')).toBeTruthy();
-  await delay(6000);
-  expect(axiosCachingDns.config.cache?.get('google.com')).toBeFalsy();
-
-  const expectedStats = {
-    dnsEntries: 0,
-    refreshed: 0,
-    hits: 0,
-    misses: 2,
-    idleExpired: 1,
-    errors: 0,
-    lastError: 0,
-    lastErrorTs: 0,
-  };
-
-  const stats = axiosCachingDns.getStats();
-  stats.refreshed = 0;
-  expect(stats).toEqual(expectedStats);
+  runTests();
 });
 
-test('query google caches and refreshes', async () => {
-  await axiosClient.get('http://google.com');
-  const { updatedTs } = axiosCachingDns.config.cache?.get('google.com');
-  const timeoutTime = Date.now() + 5000;
-  while (true) {
-    const dnsEntry = axiosCachingDns.config.cache?.get('google.com');
-    if (!dnsEntry) throw new Error('dnsEntry missing or expired');
-    if (updatedTs !== dnsEntry.updatedTs) break;
-    if (Date.now() > timeoutTime) throw new Error('Timeout exceeded');
-    await delay(10);
-  }
+describe("Redis Cache Tests", () => {
+  beforeAll(() => {
+    useRedis = true;
+    clearStats();
+  });
 
-  const expectedStats = {
-    dnsEntries: 1,
-    refreshed: 0,
-    hits: 0,
-    misses: 3,
-    idleExpired: 1,
-    errors: 0,
-    lastError: 0,
-    lastErrorTs: 0,
-  };
-
-  const stats = axiosCachingDns.getStats();
-  stats.refreshed = 0;
-  expect(stats).toEqual(expectedStats);
+  runTests();
 });
 
-test('query two services, caches and after one idle delay uncached', async () => {
-  await axiosClient.get('http://amazon.com');
-
-  await axiosClient.get('http://microsoft.com');
-  const { lastUsedTs } = axiosCachingDns.config.cache?.get('microsoft.com');
-  expect(axiosCachingDns.config.cache?.get('microsoft.com').nextIdx).toBe(1);
-
-  await axiosClient.get('http://microsoft.com');
-  expect(axiosCachingDns.config.cache?.get('microsoft.com').nextIdx).toBe(2);
-
-  expect(lastUsedTs < axiosCachingDns.config.cache?.get('microsoft.com').lastUsedTs).toBeTruthy();
-
-  expect(axiosCachingDns.config.cache?.size).toBe(2);
-  await axiosClient.get('http://microsoft.com');
-  expect(axiosCachingDns.config.cache?.get('microsoft.com').nextIdx).toBe(3);
+function runTests() {
```

Comment:
The actual tests have stayed basically the same. The major differences are:
- They're now all in this `runTests` function (for DRY code)
- We use the DNSEntryCache wrapper which can have slightly diff. methods
- They've got awaits on relevant functions (as Redis may be used)